ui/components/ui/typography/typography.js

```diff
@@ -19,15 +20,19 @@ export default function Typography({
   children,
   fontWeight = 'normal',
   align,
+  display,
   boxProps = {},
 }) {
   const computedClassName = classnames(
     'typography',
     className,
     `typography--${variant}`,
-    `typography--align-${align}`,
-    `typography--color-${color}`,
     `typography--weight-${fontWeight}`,
+    {
+      [`typography--align-${align}`]: Boolean(align),
+      [`typography--color-${color}`]: Boolean(color),
+      [`typography--display-${display}`]: Boolean(display),
```

Comment:
Ah there's no styles to wire this up in typography.scss I'm afraid, those are only on the Box component that wraps the Typography one.